test/functional/wallet_createwallet.py

```diff
@@ -113,11 +147,15 @@ def run_test(self):
         w6.keypoolrefill(1)
         # There should only be 1 key
         walletinfo = w6.getwalletinfo()
-        assert_equal(walletinfo['keypoolsize'], 1)
-        assert_equal(walletinfo['keypoolsize_hd_internal'], 1)
+        if self.options.descriptors:
+            assert_equal(walletinfo['keypoolsize'], 3)
+            assert_equal(walletinfo['keypoolsize_hd_internal'], 3)
+        else:
+            assert_equal(walletinfo['keypoolsize'], 1)
+            assert_equal(walletinfo['keypoolsize_hd_internal'], 1)
```

Comment:
Would this be a little easier to follow? Also, is the comment on l148 still relevant `# There should only be 1 key`?
```python
nkeys = 3 if self.options.descriptors else 1
assert_equal(walletinfo['keypoolsize'], nkeys)
assert_equal(walletinfo['keypoolsize_hd_internal'], nkeys)
```
```

Reply:
Done

Reply:
I thin the previous version of the code was more correct. See ci failure

Reply:
Oops, I forgot to change the second assert. Should be fixed now.